packages/client/src/__tests__/integration/happy/interactive-transactions-sqlite/test.ts

```diff
@@ -442,6 +443,38 @@ describe('interactive transactions', () => {
 
     expect(users.length).toBe(2)
   })
+
+  /**
+   * Rollback should happen even with `then` calls
+   */
+  test('rollback with then calls', async () => {
+    await prisma
+      .$transaction(async (prisma) => {
+        await prisma.user
+          .create({
+            data: {
+              email: 'user_1@website.com',
+            },
+          })
+          .then()
+
+        await prisma.user
+          .create({
+            data: {
+              email: 'user_2@website.com',
+            },
+          })
+          .then()
+          .then()
+
+        throw new Error('rollback')
+      })
+      .catch(() => {})
```

Comment:
Something like this would be nice
```suggestion
    const transaction = prisma
      .$transaction(async (prisma) => {
        await prisma.user
          .create({
            data: {
              email: 'user_1@website.com',
            },
          })
          .then()

        await prisma.user
          .create({
            data: {
              email: 'user_2@website.com',
            },
          })
          .then()
          .then()

        throw new Error('rollback')
      })
      .catch(() => {})
      await expect(transaction).rejects.toThrowErrorMatchingInlineSnapshot(``)
```